Automattic/jetpack
# PostList: initial approach

This PR creates a Proof of Concept as an approach suggested to enhance the classic view of the WordPress admin, AKA, wp-admin.

## Goals

### Share via package

It creates and exposes the whole implementation via a `automattic/jetpack-post-list` composer package (not published, yet)

It allows adding the dependency easily to the project :

```bash
composer require automattic/jetpack-post-list
```

### Use modern technologies in the UI

It uses React, Redux, etc. to rendering the UX, making the focus of the usage on @wordpress packages. It will be useful to try to keep the design parity with the block editor.

### Progressive enhancement

Although the vision of this package's goals could be super ambitious, we'd like to tackle the development/enhancement workflow in a progressively way, meaning that we could improve small sections and iterate later, again and again, reducing the risk of the introduction of errors, in favor of making the whole flow smoothly.

## Example: Posts list enhancement

This PR enhances the posts list page, specifically on the post date section.

### The approach

The app replaces the post date for each pos, into the posts list, rendering a React component that allows editing post date straight from there.
It uses the same components used in the settings section of the block editor (sidebar).

Block Editor | WPAdmin Plus / posts list page
---------|-----------
![image](https://user-images.githubusercontent.com/77539/128521250-48bdcb47-3383-41bd-82e8-132cf914042e.png) | ![image](https://user-images.githubusercontent.com/77539/128521465-f4cd7eae-06d0-4813-97b0-24f6c76dc471.png)


## Video
https://user-images.githubusercontent.com/77539/128521293-68ff151f-c694-4938-bf27-729018ed8e3e.mov


#### Changes proposed in this Pull Request:

*

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



### Testing instructions:






* checkout this branch

```bash
git checkout update/add-wp-admin-plus-package
```

* Jetpack dev-env . Add this dependency to Jetpack plugin

```bash
cd projects/plugins/jetpack && composer require automattic/jetpack-post-list
```

* wait for the dependencies update process. You should see something like the following
<img src="https://user-images.githubusercontent.com/77539/128534806-5cbd89c1-924b-4271-93b8-f222278f2410.png" width="500px" />

* Update pnpm just in case 🤞 

```bash
pnpm install -f
```

* Update Jetpack dependencies

```bash
jetpack install -r && jetpack install --all
```

* Run posts-list package compiling process

```bash
jetpack build packages/post-list
```

Use `watch` in case you'd like to dev editing the code in the client

```bash
jetpack watch packages/post-list
```

* Run web server

```
jetpack docker up
```

* Testing in a client / browser
* 
Load the posts admin page in the wp-admin, defining the `post-list=true` query variable:

`http://localhost/wp-admin/edit.php?post-list=true`

Confirm you can edit the post date by clicking on the date button:
<img src="https://user-images.githubusercontent.com/77539/128535812-d030ae94-b3d0-4a77-a3b3-0a2975890cbc.png" width="400px" />

## Tasks

- [x] Rename package to post-list. [more](https://github.com/Automattic/jetpack/pull/20589#issuecomment-896766000)
- [ ] Improve updating post date value. Consider do it when picker loses focus, `Undo` option, etc. [more](https://github.com/Automattic/jetpack/pull/20589#issuecomment-896766000)
- [ ] Show update message: when the date, status, etc changes. [more](https://github.com/Automattic/jetpack/pull/20589#issuecomment-896766000)
- [ ] Consider we might want to have a publish/draft toggle. [more](https://github.com/Automattic/jetpack/pull/20589#issuecomment-896766000)
- [ ] Improve experience resetting the calendar date value. [more](https://github.com/Automattic/jetpack/pull/20589#issuecomment-896766000)
